src/ostorlab/cli/utils.py

```diff
@@ -0,0 +1,88 @@
+"""Module responsible for fetching the agent details from the container image."""
+
+import logging
+import io
+from typing import Any
+
+import click
+import docker
+import docker.errors
+
+from ostorlab import configuration_manager
+from ostorlab.apis import agent_details as agent_details_api
+from ostorlab.apis.runners import public_runner, authenticated_runner
+from ostorlab.apis.runners import runner as base_runner
+from ostorlab.cli import console as cli_console
+from ostorlab.agent import definitions as agent_definitions
+
+logger = logging.getLogger(__name__)
+
+console = cli_console.Console()
+
+
+class Error(Exception):
+    """Base Error."""
+
+
+class AgentDetailsNotFound(Error):
+    """Agent not found error."""
+
+
+def get_agent_details(agent_key: str) -> dict[Any, Any]:
+    """Sends an API request with the agent key, and retrieve the agent information.
+
+    Args:
+        agent_key: the agent key in the form : agent/org/name
+
+    Returns:
+        dictionary of the agent information like : name, dockerLocation..
+
+    Raises:
+        click Exit exception with satus code 2 when API response is invalid.
+    """
+    config_manager = configuration_manager.ConfigurationManager()
+
+    if config_manager.is_authenticated:
+        runner = authenticated_runner.AuthenticatedAPIRunner()
+    else:
+        runner = public_runner.PublicAPIRunner()
+
+    try:
+        response = runner.execute(agent_details_api.AgentDetailsAPIRequest(agent_key))
+    except base_runner.ResponseError as e:
+        raise AgentDetailsNotFound("requested agent not found") from e
+
+    if "errors" in response:
+        error_message = f"""The provided agent key : {agent_key} does not correspond to any agent.
+        Please make sure you have the correct agent key.
+        """
+        raise AgentDetailsNotFound(error_message)
+    else:
+        agent_details = response["data"]["agent"]
+        return agent_details
+
+
+def get_agent_definition(
+    agent_key: str,
+) -> agent_definitions.AgentDefinition:
+    """
+    Fetch args of an agent from container image.
+    """
```

Comment:
Can you update the doctstring to include `agent_key` as well as what the function returns?